components/misc/LikeButton.tsx

```diff
@@ -3,41 +3,46 @@ import { useEffect, useState } from 'react';
 import IconToggle from './IconToggle';
 import { Ionicons } from '@expo/vector-icons';
 import { DebounceSession } from '../../utils/utils';
+import { User } from '../../xplat/types/types';
+import { useRecoilValue } from 'recoil';
+import { userAtom } from '../../utils/atoms';
 
 type Props = {
-  isLiked: boolean;
-  onSetLiked: (isLiked: boolean) => void;
-  numLikes: number;
+  likes: User[];
+  onSetIsLiked: (isLiked: boolean) => void;
 };
-const LikeButton = ({ isLiked, onSetLiked, numLikes }: Props) => {
-  const [isLikedLocal, setIsLikedLocal] = useState<boolean>(isLiked);
-  const [numLikesLocal, setNumLikesLocal] = useState<number>(numLikes);
+const LikeButton = ({ likes, onSetIsLiked }: Props) => {
+  const user = useRecoilValue(userAtom);
+
+  const [userIsInLikes, setUserIsInLikes] = useState<boolean>(false);
+  const [isLiked, setIsLiked] = useState<boolean>(false);
   const [debounceSession] = useState<DebounceSession>(
     new DebounceSession(1000)
   );
 
   useEffect(() => {
-    setIsLikedLocal(isLiked);
-    setNumLikesLocal(numLikes);
-  }, [isLiked, numLikes]);
+    if (user === undefined) return;
+    const _userIsInLikes = likes.some(
+      (like) => like.docRef!.id === user.docRef!.id
+    );
+    setUserIsInLikes(_userIsInLikes);
```

Comment:
If these are always the same... then why have two different things

Reply:
Sorry I don't understand the question. What are you referring to with "these"

Reply:
Oh I see, you didn't highlight the whole block you were talking about in the comment. The state is initialized to the same, and then only one changes on like. This is so that the `isLiked` is actually meaningful in relation to the length of the likes array.